### DISCUSSION TITLE
Expose custom jmx port for connector metrics

### DISCUSSION THREAD

**Discussion by cyberjar09** [strimzi/strimzi-kafka-operator#7645] [Q&A] [answered]
- I am using debezium mongodb source connector 
- https://debezium.io/documentation/reference/stable/connectors/mongodb.html#mongodb-monitoring
- there are some metrics specific to this connector which can be emitted via JMX
- by setting `jmxOptions{}` the kafka-connect metrics are available but I need to expose a separate port on the service/pod which can be scraped to get the debezium metrics too
- I the closest I got by browsing the docs was to use the [InternalServiceTemplate](https://strimzi.io/docs/operators/latest/configuring.html#type-InternalServiceTemplate-reference) but that does not expose any way for me to add a new port
- so kinda lost now, need help please 🙏

**Comment by scholzj** [comment, marked as answer]
You cannot expose any custom ports. However, it is not really clear to me why would you need it in this case ...

----

Is that really how JMX works? That there are multiple different "JMX" subsystems? One for Connect and one for Connector? I'm not really JMX expert ... but I have a some doubts about it. I always assumed the JMX subsystem is JVM-wide. At the end, it is not the application which exposes the JMX. It is the JVM.

If it would be the case that you need a separate port, how would the JMX be actually exposed on the "new" port? Can you share some examples of how would the commands to start Connect look like with multiple JMX listeners?

**Reply by cyberjar09** [reply]
- hmmm, thats a great point actually. 
- Im very far from a JMX expert (nodejs dev by day). 
- My mental model was more aligned with how a process can bind multiple host OS ports to "servers". 
- So kafka-connect exposes a JMX server on 9999 but debezium JMX server could use 5555 ... is what I Was thinking
  - seems flawed now
- I was led down this path because when I tried to expose the debezium metrics on the same port, I was able to see the kafka-connect metrics but not debezium 😕 
- so now im left scratching my head again

**Reply by scholzj** [reply]
I'm not sure where did you come up with port 5555. In general, you can expose JMX on any free port. In Strimzi, we use 9999. If someone else somewhere else uses 5555, that is fine, but it does not mean anything for Strimzi since we use 9999.
